src/main/java/de/tum/in/www1/artemis/web/rest/ParticipationResource.java

```diff
@@ -215,21 +210,20 @@ public ParticipationResource(ParticipationService participationService, Programm
             participation = participationService.resumeExercise(participation);
             // Note: in this case we might need an empty commit to make sure the build plan works correctly for subsequent student commits
             participation = participationService.performEmptyCommit(participation);
-            if (participation != null) {
-                addLatestResultToParticipation(participation);
+            addLatestResultToParticipation(participation);
+            participation.getExercise().filterSensitiveInformation();
+            return ResponseEntity.ok().headers(HeaderUtil.createEntityUpdateAlert(applicationName, true, ENTITY_NAME, participation.getStudent().getName())).body(participation);
+        }
+        else {
```

Comment:
Only in the case if participation is null, which means that exercise attached to the participation is not of Programming Exercise type we reach this part

Reply:
revert

Reply:
Done